src/main/java/org/folio/circulation/storage/SingleOpenLoanByUserAndItemBarcodeFinder.java

```diff
@@ -44,13 +48,15 @@
     final UserNotFoundValidator userNotFoundValidator = new UserNotFoundValidator(
       userId -> failure("user is not found", "userId", userId));
 
+    final BlockRenewalValidator blockRenewalValidator =
+      new BlockRenewalValidator(requestQueueRepository);
+
     return requestResult
       .after(checkInRequest -> itemFinder.findItemByBarcode(itemBarcode))
-      .thenComposeAsync(itemResult ->
-        itemResult.after(singleOpenLoanFinder::findSingleOpenLoan))
+      .thenComposeAsync(itemResult -> itemResult.after(blockRenewalValidator::refuseWhenFirstRequestIsRecall))
```

Comment:
Given this is used by both renewal and overridden renewal, this may need to change if this check should not apply to overrides.

Reply:
Cate is fine with it, but Sean is still silent...